dotnetv3/Glue/Tests/GlueTests.cs

```diff
@@ -0,0 +1,238 @@
+            }
+            while (crawlerState != "READY");
+
+            Assert.True(success, "Could not create crawler.");
+        }
+
+        /// <summary>
+        /// Test starting an AWS Glue crawler.
+        /// </summary>
+        /// <returns>An async Task.</returns>
+        [Fact]
+        [Order(2)]
+        [Trait("Category", "Integration")]
+        public async Task StartCrawlerAsyncTest()
+        {
+            var success = await _wrapper.StartCrawlerAsync(_crawlerName);
+
+            // Wait for the crawler to be started.
+            CrawlerState crawlerState;
+            do
+            {
+                crawlerState = await _wrapper.GetCrawlerStateAsync(_crawlerName);
+            }
+            while (crawlerState != "READY");
+
+            Assert.True(success, "Could not start crawler.");
+        }
+
+        /// <summary>
+        /// Test getting AWS Glue database information.
+        /// </summary>
+        /// <returns>An async Task.</returns>
+        [Fact]
+        [Order(3)]
+        [Trait("Category", "Integration")]
+        public async Task GetDatabaseAsyncTest()
+        {
+            var database = _wrapper.GetDatabaseAsync(_dbName);
+            Assert.NotNull(database);
+        }
+
+        /// <summary>
+        /// Test getting information about AWS Glue database tables.
+        /// </summary>
+        /// <returns>An async Task.</returns>
+        [Fact]
+        [Order(4)]
+        [Trait("Category", "Integration")]
+        public async Task GetTablesAsyncTest()
+        {
+            _tables = await _wrapper.GetTablesAsync(_dbName);
+            Assert.NotEmpty(_tables);
+        }
+
+        /// <summary>
+        /// Test creating an AWS Glue job.
+        /// </summary>
+        /// <returns>An async Task.</returns>
+        [Fact]
+        [Order(5)]
+        [Trait("Category", "Integration")]
+        public async Task CreateJobAsyncTest()
+        {
+            var success = await _wrapper.CreateJobAsync(_jobName, _roleName, _description, _scriptUrl);
+            Assert.True(success, "Couldn't create job.");
+        }
+
+        /// <summary>
+        /// Test starting an AWS Glue job run.
+        /// </summary>
+        /// <returns>An async Task.</returns>
+        [Fact]
+        [Order(6)]
+        [Trait("Category", "Integration")]
+        public async Task StartJobRunAsyncTest()
+        {
+            _jobRunId = await _wrapper.StartJobRunAsync(_jobName);
+
+            // Wait for the job run to complete or error out.
+            var jobRunComplete = false;
+            do
+            {
+                var jobRun = await _wrapper.GetJobRunAsync(_jobName, _jobRunId);
+                if (jobRun.JobRunState == "SUCCEEDED" || jobRun.JobRunState == "STOPPED" ||
+                    jobRun.JobRunState == "FAILED" || jobRun.JobRunState == "TIMEOUT")
+                {
+                    jobRunComplete = true;
+                }
+            } while (!jobRunComplete);
```

Comment:
Is this test going to pass even if the job run fails?

Reply:
Yes. It will pass because I'm testing for an end to the job run including the run ending for errors.